include/deal.II/matrix_free/matrix_free.h

```diff
@@ -4652,11 +4652,34 @@ namespace internal
           }
     }
 
-    // Runs the assembler on interior faces. If no function is given, nothing
-    // is done
     virtual void
-    face(const std::pair<unsigned int, unsigned int> &face_range) override
+    cell(const unsigned int range_index) override
```

Comment:
@kronbichler  The PR is not ready yet. But I would like to have your opinion on the approach, since it changes the internal interface of `MFWorkerInterface`. I would like to pass here a `range_index`/`partition_index` that allows me later to efficiently to access some new vectors in `TaskInfo` (containing the pre-computed info).

Reply:
I fully agree with this change, so please go ahead. This is internal and just used for driving the loops, so I see no need to stick with an interface. One thing I wonder is whether the other `cell` function is still needed or if we can route everything in the new function.